src/sagemaker/jumpstart/types.py

```diff
@@ -1253,18 +1115,1090 @@ def from_json(self, json_obj: Dict[str, Any]) -> None:
+
+            self.training_script_uri: Optional[str] = json_obj.get("TrainingScriptUri")
+            self.training_prepacked_script_uri: Optional[str] = json_obj.get(
+                "TrainingPrepackedScriptUri"
+            )
+            self.training_prepacked_script_version: Optional[str] = json_obj.get(
+                "TrainingPrepackedScriptVersion"
+            )
+            self.training_ecr_uri: Optional[str] = json_obj.get("TrainingEcrUri")
+            self._non_serializable_slots.append("training_ecr_specs")
+            self.training_metrics: Optional[List[Dict[str, str]]] = json_obj.get(
+                "TrainingMetrics", None
+            )
+            self.training_artifact_uri: Optional[str] = json_obj.get("TrainingArtifactUri")
+            self.training_dependencies: Optional[str] = json_obj.get("TrainingDependencies")
+            self.default_training_instance_type: Optional[str] = json_obj.get(
+                "DefaultTrainingInstanceType"
+            )
+            self.supported_training_instance_types: Optional[str] = json_obj.get(
+                "SupportedTrainingInstanceTypes"
+            )
+            self.training_volume_size: Optional[int] = json_obj.get("TrainingVolumeSize")
+            self.training_enable_network_isolation: Optional[str] = json_obj.get(
+                "TrainingEnableNetworkIsolation", False
+            )
+            self.training_instance_type_variants: Optional[JumpStartInstanceTypeVariants] = (
+                JumpStartInstanceTypeVariants(
+                    json_obj.get("TrainingInstanceTypeVariants"),
+                    is_hub_content=True,
+                )
+                if json_obj.get("TrainingInstanceTypeVariants")
+                else None
+            )
+            # Estimator kwargs
+            self.encrypt_inter_container_traffic: Optional[bool] = (
+                bool(json_obj.get("EncryptInterContainerTraffic"))
+                if json_obj.get("EncryptInterContainerTraffic")
+                else None
+            )
+            self.max_runtime_in_seconds: Optional[str] = json_obj.get("MaxRuntimeInSeconds")
+            self.disable_output_compression: Optional[bool] = (
+                bool(json_obj.get("DisableOutputCompression"))
+                if json_obj.get("DisableOutputCompression")
+                else None
+            )
+
+    def get_schema_version(self) -> str:
+        """Returns schema version."""
+        return self.SCHEMA_VERSION
+
+    def get_region(self) -> str:
+        return self._region
+
+    def from_specs(self, model_specs: JumpStartModelSpecs, studio_specs: Dict[str, Any]) -> None:
+        self.url: str = model_specs.url
+        self.min_sdk_version: str = model_specs.min_sdk_version
+        self.hosting_ecr_specs: Optional[JumpStartECRSpecs] = model_specs.hosting_ecr_specs
+        self._non_serializable_slots.append("hosting_ecr_uri")
+
+        content_bucket = f"jumpstart-cache-prod-{self._region}"
```

Comment:
This won't be the case for gated models- can you use the same logic as here? 

https://github.com/aws/sagemaker-python-sdk/blob/master-jumpstart-curated-hub/src/sagemaker/jumpstart/curated_hub/accessors/public_model_data.py#L41-L44

Reply:
also, we have a function that gets the content bucket, do not construct the string manually